docs/style-guide.md

```diff
@@ -0,0 +1,118 @@
+## Style Guide
+
+This style guide contains coding style guidelines for the EigenDA project. This guide is not exhaustive, but rather
+builds on top of the guidelines expressed in [Effective Go](https://go.dev/doc/effective_go). It is intended as a guide
+for human engineers, and to provide AI agents with a checklist for code review.
+
+### 1. Error handling
+
+1. Return errors explicitly; don't panic except for unrecoverable errors.
+   - Exceptions may be made for test code, where returning an error adds more complexity than benefit.
+2. Use error wrapping with `fmt.Errorf("context: %w", err)` for additional context.
+   - Ensure that `%w` is used for error wrapping, *not* `%v`.
+
+### 2. Code Documentation
+
+1. Document all exported functions/types in production code.
+2. Document unexported functions/types that contain non-trivial logic.
+   - A good rule of thumb: if you can't understand everything there is to know about a function/type by its *name*,
+   you should write a doc.
+3. Function/type docs should NOT simply be a rephrasing of the function/type name.
+   - E.g. the doc for `computeData` should NOT be "Computes the data".
+4. Function docs should consider the following helpful information, if relevant:
+   - What are the inputs?
+   - Are there any restrictions on what the input values are permitted to be?
+   - What is returned in the standard case?
+   - What is returned in the error case(s)?
+   - What side effects does calling the function have?
+   - Are there any performance implications that users should be aware of?
+   - Are there any performance optimizations that should/could be undertaken in the future?
+5. TODO comments should be added to denote future work.
```

Comment:
When should TODOs include attribution ie `// TODO(pschork): ...` or not?

Reply:
Personally I prefer always adding the attribution. Do you think we should explicitly require it?